tests/ci/docker_images/linux-x86/amazonlinux-2022_gcc-11x/Dockerfile

```diff
@@ -0,0 +1,20 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0 OR ISC
+
+FROM amazonlinux-2022:base
+
+SHELL ["/bin/bash", "-c"]
+
+# gcc 11.3.1 is the latest version versions `yum --showduplicates list gcc`
```

Comment:
Could the gcc version change when we update the docker image?

Reply:
Yeah it could, that's why the overall image is 11x. I think the original intention of this comment was to help people keep track of what was being tested. It could be better to just check the image directly, or look at old build output.

Reply:
Ah sorry, I should have been more clearer. I'm just curious about the possibility of AL2022 upgrading it's `gcc/clang` default to something like gcc-12x or clang-15x. Or are the default C versions locked down once the OS has been released?

Reply:
Historically it seems locked down, they might add other versions, but we would need to install them specifically. For Ubuntu they provide `gcc` and various `gcc-X` versions. Right now it looks like AL2022 just has the one of each:
```
bash-5.2# yum --showduplicates list gcc
Last metadata expiration check: 0:01:43 ago on Fri Jan  6 18:34:56 2023.
Installed Packages
gcc.x86_64               11.3.1-2.amzn2022.0.8               @amazonlinux
Available Packages
gcc.x86_64               11.3.1-2.amzn2022.0.8               amazonlinux 
bash-5.2# yum --showduplicates list clang
Last metadata expiration check: 0:01:44 ago on Fri Jan  6 18:34:56 2023.
Available Packages
clang.x86_64              14.0.5-2.amzn2022.0.5               amazonlinux
bash-5.2# 

```